### ISSUE TITLE
Reuse error from the loss function

### ISSUE THREAD

**Issue by David-J-R** [David-J-R/BEclear#12]
The Error, difference  between D - L*R is already calculated in the loss function, but is than calculated again during the gradient descent. 
Saving the Error should save time. Implement this after issue #11
